File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/C++/_043)_Find_Equilibrium_Point_in_an_Array.cpp
#include<iostream>
using namespace std;

bool checkEquilibrium(int arr[], int n)
{
	int sum = 0;
	for(int i = 0; i < n; i++)
		sum += arr[i];
	int l_sum = 0;
	for(int i = 0; i < n; i++)
	{
		if(l_sum == sum - arr[i])
			return true;
		l_sum += arr[i];
		sum -= arr[i];
	}
	return false;
}

int main()
{
    int arr[] = {4, 2, 2}, n = 3;
    cout<<checkEquilibrium(arr, n)? "true" : "false";
    cout<<endl;
}

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_043)_Find_Equilibrium_Point_in_an_Array.dart
bool checkEquilibrium(List<int> arr) {
  int sum = 0;

  for (int i = 0; i < arr.length; i++) sum += arr[i];

  int lSum = 0;

  for (int i = 0; i < arr.length; i++) {
    if (lSum == sum - arr[i]) return true;
    lSum += arr[i];
    sum -= arr[i];
  }

  return false;
}

void main() {
  var arr = [3, 4, 8, -9, 20, 6];
  print('Equilibrium = ${checkEquilibrium(arr)}');
}
